### PR TITLE
Implement __sizeof__ function

### LINKED ISSUE #444: Implement __sizeof__ function to correctly calculate all internal structures size.
See also https://docs.python.org/3/library/sys.html#sys.getsizeof

### PR DESCRIPTION
**Pull request by asvetlov**
Fixes #444

### CODE REVIEW

File: multidict/_multidict.c [hunk 1]
```diff
@@ -982,6 +999,12 @@ static PyMethodDef multidict_methods[] = {
         METH_O | METH_CLASS,
         NULL
     },
+    {
+        "__sizeof__",
+        _multidict_sizeof,
```
Comment (iemelyanov) [inline multidict/_multidict.c]: `(PyCFunction)_multidict_sizeof` more explicit

File: multidict/_multidict.c [hunk 2]
```diff
@@ -863,6 +864,22 @@ multidict_class_getitem(PyObject *self, PyObject *arg)
     return self;
 }
 
+
+PyDoc_STRVAR(sizeof__doc__,
+"D.__sizeof__() -> size of D in memory, in bytes");
+
+static inline PyObject *
```
Comment (iemelyanov) [inline multidict/_multidict.c]: `inline` is not necessary

File: multidict/_multidict.c [hunk 3]
```diff
@@ -863,6 +864,22 @@ multidict_class_getitem(PyObject *self, PyObject *arg)
...
+_multidict_sizeof(PyObject *self)
```
Comment (iemelyanov) [inline multidict/_multidict.c]: Mb more correct `_multidict_sizeof(PyObject *self, PyObject *args)` according to `PyCFunction` and `PyMethodDef` ?

File: multidict/_multidict.c (hunk 2 above)
Comment (asvetlov) [reply multidict/_multidict.c]: Yes, I know.
I've converted all functions to inlines after adding `_multilib`, let's keep this style at least for a while. It doesn't hurt.

File: multidict/_multidict.c (hunk 3 above)
Comment (asvetlov) [reply multidict/_multidict.c]: We use `METH_O` convention in other places.
I've updated the PR to accept `MultiDictObject * self` as an argument.

**Comment by asvetlov** [comment]
Thanks for the review!

**Pull request merged**